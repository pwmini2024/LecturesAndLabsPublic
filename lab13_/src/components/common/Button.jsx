import React from "react";

function Button({ onClick, children, disabled }) {
  return (
    <button
      className={`custom-button ${disabled ? "disabled" : ""}`}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
}

export default Button;
