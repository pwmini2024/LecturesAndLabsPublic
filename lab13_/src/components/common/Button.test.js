import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Button from "./Button";

// Unit Tests for Button Component
describe("Button Component", () => {
  // Snapshot test
  it("matches snapshot", () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container).toMatchSnapshot();
  });

  // Props test
  it("renders button text correctly", () => {
    render(<Button>Click me</Button>);
    expect(screen.getByText("Click me")).toBeInTheDocument();
  });

  // Event handler test
  it("calls onClick handler when clicked", () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click me</Button>);

    fireEvent.click(screen.getByText("Click me"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  // Disabled state test
  it("is disabled when disabled prop is true", () => {
    render(<Button disabled>Click me</Button>);
    expect(screen.getByText("Click me")).toBeDisabled();
  });

  // Class name test
  it("applies correct classes", () => {
    const { container } = render(<Button disabled>Click me</Button>);
    expect(container.firstChild).toHaveClass("custom-button", "disabled");
  });
});
