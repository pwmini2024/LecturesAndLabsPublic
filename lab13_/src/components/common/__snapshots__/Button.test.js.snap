// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<div>
  <button
    class="custom-button "
  >
    Click me
  </button>
</div>
`;
